presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1211,6 +1271,28 @@ private Partition buildPartitionObject(ConnectorSession session, Table table, Pa
                 .build();
     }
 
+    private PartitionStatistics createPartitionStatistics(
+            ConnectorSession session,
+            Map<String, Type> columnTypes,
+            ComputedStatistics computedStatistics)
+    {
+        Map<ColumnStatisticMetadata, Block> computedColumnStatistics = computedStatistics.getColumnStatistics();
+
+        Block rowCountBlock = Optional.ofNullable(computedStatistics.getTableStatistics().get(ROW_COUNT))
+                .orElseThrow(() -> new IllegalArgumentException("rowCount not present"));
+        verify(!rowCountBlock.isNull(0), "rowCount must never be null");
+        long rowCount = BIGINT.getLong(rowCountBlock, 0);
+
+        Map<String, HiveColumnStatistics> columnStatistics = fromComputedStatistics(
+                session,
+                timeZone,
+                computedColumnStatistics,
+                columnTypes,
+                rowCount);
+        HiveBasicStatistics rowCountOnlyBasicStatistics = new HiveBasicStatistics(OptionalLong.empty(), OptionalLong.of(rowCount), OptionalLong.empty(), OptionalLong.empty());
```

Comment:
I think we should recompute the basic statistics as well. `fileCount`, `inMemoryDataSizeInBytes`, `onDiskDataSizeInBytes` are important for us. We don't want ANALYZE statement to reset those. `fileCount` and `onDiskDataSizeInBytes` are simple. 

You just go to the file system, and scan the file under the partition directory. `inMemoryDataSizeInBytes` is a little bit more complex. You need to add an aggregation function that will sum up the data size for the entire row. Currently it is not possible to add such aggregation in our framework.

So, i would suggest you to:

1. In this commit update `SemiTransactionalHiveMetastore` to do not override these statistics it ones are present.
2. As a separate commit add the `fileCount` and `onDiskDataSizeInBytes` statistics computation.
3. As a separate PR improve the aggregation framework, so it is supports either row wide aggregation, or variadic bound aggregation. (Not sure what is better. @haozhun @wenleix ?)Create an aggregation function that will sum up the page sizes, and compute this statistic as well.